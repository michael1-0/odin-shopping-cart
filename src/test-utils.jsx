import { render } from "@testing-library/react";
import { RouterProvider, createMemoryRouter } from "react-router";
import routes from "./routes"; 

export function renderWithRouter() {
  const router = createMemoryRouter(routes);

  return {
    ...render(<RouterProvider router={router} />),
    router,
  };
}
