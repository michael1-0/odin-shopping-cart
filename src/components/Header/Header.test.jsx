import { it, describe, expect } from "vitest";
import { screen } from "@testing-library/react";
import { renderWithRouter } from "../../test-utils";

describe("Header", () => {
  it("should match snapshot", () => {
    const { container } = renderWithRouter();
    expect(container).toMatchSnapshot();
  });

  it("should render correct header links", () => {
    renderWithRouter();

    expect(screen.getByRole("link", { name: /home/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /shop/i })).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /cart/i })).toBeInTheDocument();
  });
});
