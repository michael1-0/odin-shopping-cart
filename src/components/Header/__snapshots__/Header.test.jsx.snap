// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header > should match snapshot 1`] = `
<div>
  <div
    class="wrapper"
  >
    <header
      class="_header_a5b977"
    >
      <div>
        0 item in cart
      </div>
      <nav
        class="_nav_a5b977"
      >
        <a
          data-discover="true"
          href="/"
        >
          Home
        </a>
        <a
          data-discover="true"
          href="/shop"
        >
          Shop
        </a>
        <a
          data-discover="true"
          href="/cart"
        >
          Cart
        </a>
      </nav>
    </header>
    <main>
      <div
        class="_wrapper_650bde"
      >
        <div>
          <h1>
            Home
          </h1>
          <p>
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Ea, nemo quos aliquam, facere ad sit laborum rerum perspiciatis doloremque autem placeat fuga alias sint ullam ratione rem veniam nisi saepe!
          </p>
        </div>
        <svg
          fill="currentColor"
          height="500"
          viewBox="0 0 16 16"
          width="500"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M0 1.5A.5.5 0 0 1 .5 1H2a.5.5 0 0 1 .485.379L2.89 3H14.5a.5.5 0 0 1 .49.598l-1 5a.5.5 0 0 1-.465.401l-9.397.472L4.415 11H13a.5.5 0 0 1 0 1H4a.5.5 0 0 1-.491-.408L2.01 3.607 1.61 2H.5a.5.5 0 0 1-.5-.5M3.102 4l.84 4.479 9.144-.459L13.89 4zM5 12a2 2 0 1 0 0 4 2 2 0 0 0 0-4m7 0a2 2 0 1 0 0 4 2 2 0 0 0 0-4m-7 1a1 1 0 1 1 0 2 1 1 0 0 1 0-2m7 0a1 1 0 1 1 0 2 1 1 0 0 1 0-2"
          />
        </svg>
      </div>
    </main>
    <footer
      class="_footer_acfd2c"
    >
       $
      0.00
       to be paid 
    </footer>
  </div>
</div>
`;
